### PR TITLE
Add the ability to better pass additional arguments to containers

### PR DESCRIPTION
**Pull request by Rajat-0**
#1540

### CODE REVIEW

**Comment by Rajat-0** [comment]
@mwielgus I was trying to implement https://github.com/kubernetes/autoscaler/issues/1540

**Comment by mwielgus** [comment]
OK, I noticed the bug in the description, but I'm not sure what is your plan in modifying the build scripts.

**Comment by Rajat-0** [comment]
@mwielgus So if I have to customize the arguments to be passed to the containers, then I have to change in Its pod's DockerFile( Recommender, Admission Controller, Updater)

**Comment by mwielgus** [comment]
You are changing the wrong Docker file. For example you should change this one:

https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/Dockerfile

Instead of the one that is used to compile stuff.

File: vertical-pod-autoscaler/pkg/updater/Dockerfile [hunk 1]
```diff
@@ -18,4 +18,4 @@ MAINTAINER Marcin Wielgus "mwielgus@google.com"
 
 ADD updater updater
 
-CMD ./updater --v=4 --stderrthreshold=info
+ENTRYPOINT ./updateri --v=4 --stderrthreshold=info
```
Comment (mwielgus) [inline vertical-pod-autoscaler/pkg/updater/Dockerfile]: typo

**Comment by mfilotto** [comment]
Maybe we could move default parameters to CMD as a best practice
https://medium.freecodecamp.org/docker-entrypoint-cmd-dockerfile-best-practices-abc591c30e21

And there is an extra i at the end of updater

File: vertical-pod-autoscaler/pkg/updater/Dockerfile (hunk 1 above)
Comment (Rajat-0) [reply vertical-pod-autoscaler/pkg/updater/Dockerfile]: Resolved

**Comment by mfilotto** [comment]
Any plan to build a new vpa images version to unlock this feature ?
Looking forward to be able to set arguments (set replicas to 1 in my use case)

**Comment by Leluke** [comment]
Hi how can I take advantage of this option? I wanted my VPA to restart pods even when there is only one replica of each.

**Pull request merged**